core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -66,6 +65,9 @@ private[spark] class CoarseMesosSchedulerBackend(
 
   val extraCoresPerSlave = conf.getInt("spark.mesos.extra.cores", 0)
 
+  // Offer constraints
+  val slaveOfferConstraints = parseConstraintString(sc.conf.get("spark.mesos.constraints", ""))
```

Comment:
Nit: Let's use the same exact style in both coarse and fine grain mode:

val slaveOfferConstraints = parseConstraintString(sc.conf.get("spark.mesos.constraints", ""))

vs

private[mesos] val constraintsStr = sc.conf.get("spark.mesos.constraints", "")
private[mesos] val offerConstraints = parseConstraintString(constraintsStr)

How about just use the top style?